### PR TITLE
Add "protect from ignores" option per-NPC, closes #377

### LINKED ISSUE #377: <feature> Make the "random.protected" config option configurable per-NPC
Currently, the config option "random.protected" can only be changed globally, there is no way to make a specific NPC protected from ignored targets.
I propose to add a "/sentinel protectfromignores" command to allow this functionnality.

### PR DESCRIPTION
**Pull request by SkytAsul**
* added /sentinel protectFromIgnores [true/false]
* same behavior as the "random.protected" config option

### CODE REVIEW

**Comment by mcmonkey4eva** [comment]
- Should be labeled `protected` (same as current config option, and also just ... not so awkwardly long nor mixed case)
- Needs a default value in the `sentinel defaults` section of the config
- Needs to be in the readme

File: src/main/java/org/mcmonkey/sentinel/commands/SentinelTargetCommands.java [hunk 1]
```diff
@@ -263,6 +264,27 @@ public void runaway(CommandContext args, CommandSender sender, SentinelTrait sen
         }
     }
 
+    @Command(aliases = {"sentinel"}, usage = "protectFromIgnores ['true'/'false']",
+            desc = "Toggles whether the NPC cannot be harmed by ignored targets.",
+            modifiers = {"protectFromIgnores"}, permission = "sentinel.fightback", min = 1, max = 2)
```
Comment (mcmonkey4eva) [inline src/main/java/org/mcmonkey/sentinel/commands/SentinelTargetCommands.java:269]: wrong permission node there

**Comment by SkytAsul** [comment]
> * Should be labeled `protected` (same as current config option, and also just ... not so awkwardly long nor mixed case)
`protected` is a reserved word in Java. And I think `protected` is not really explanatory of what is the option doing.

> Needs a default value in the `sentinel defaults` section of the config
Alright bu in this case, what should we do about the existing "protected" option? Let it here and have two different config options for the same thing? I find it... weird.

**Comment by mcmonkey4eva** [comment]
the field name can be different (as it is for the existing one) but the user-exposed name should be the shorter version.

**Comment by SkytAsul** [comment]
Renamed it to "/sentinel protected". However the command cannot have a Java-reserved word so I put in the modifiers, I do not know if it will work.

**Pull request merged**